### DISCUSSION TITLE
Enhancement: suggest changes on articels and comments of other users

### DISCUSSION THREAD

**Discussion by zukunft** [orbiting/republik-frontend#545] [Ideas]
Allow to suggest changes on articles and comments of other users. 

Some users seem more typo mistakes the other and they [like to](https://www.republik.ch/dialog?id=6cb76f52-31ac-413b-83d3-03e235f13de8&t=article&focus=a949bbc9-4f68-44fb-a7cc-de628e561600) suggest the corrections to the author.  The suggested setup is:

1. Should **all users** have the possibility to suggest changes: **No**, because this would make the interface more complex and most of the users are not expected to use this function.
2. Should the corrector **role** be **assigned by** the **admin** only: **No**, because the correction should not be live directly. Instead the author should receive a message with the suggested changes.
3. Should the author have the possibility to **confirm change by change**: **Yes**, because it is expected that not so many correction will be suggested. 

Steps to implement:

1. Add a new **role "corrector"** if it does not yet exists and allow each user to switch on the role
2. If the user has the corrector role, **all articles* and comments have the **change icon**
3. if an article or comment is changed and the changer is not the author, the changes are detected change by change and **send to** the **author** as a single change request
4. in the inbox of the author the suggested changes can be **confirmed** change by change and once confirmed the article or comment is changed according to the suggestion
5. The author can also **reject** a change and **comment**, why he rejected it. This message is send to the inbox of the corrector.

The change message should contain a link to the article or comment, the sentence with the suggested change, plus one sentence before and after. The author should have a selector to confirm or reject the change and a text field for a comment. All suggested changes are shown in a paged list and with a confirm button the accepted or rejected changes are archived and show in a separate list.

**Comment by tpreusse** [comment]
Thank you for this idea @zukunft 

I think suggesting corrections is definitely something we should explore. I personally would imagine a simpler solution initially for articles: a small report error button at the end for all signed in users. But eventually we should make it more elaborate and consider it for the dialog too.
